### DISCUSSION TITLE
VSCode Extension for GPT-Engineer leveraging Websockets

### DISCUSSION THREAD

**Discussion by zdanl** [AntonOsika/gpt-engineer#863] [Ideas]
@keosariel https://github.com/keosariel/ just dropped this idea on me, and i think it deserves an own thread.

**Comment by zdanl** [comment]
especially for improving and editing existing code bases, this would be amazing! 

and increases visibility and prominency of the project, through the VSCode extension marketplace

**Comment by keosariel** [comment]
Having a Visual Studio Code extension would be beneficial, with the server active locally, queries could be executed directly from VS Code, and the code structure would be generated within the same environment. This approach allows for convenient testing and editing of the program directly within the development environment.

**Comment by zdanl** [comment]
hopefully useful:

https://code.visualstudio.com/api/get-started/your-first-extension

https://github.com/microsoft/vscode-extension-samples
